terminus-ui/autocomplete/src/autocomplete.component.ts

```diff
@@ -120,152 +231,228 @@ export class TsAutocompleteComponent<OptionType = {[name: string]: any}> impleme
+  @ContentChildren(TsOptionComponent, { descendants: true })
+  public options!: QueryList<TsOptionComponent>;
+
+  /**
+   * Access all of the defined groups of options
+   */
+  @ContentChildren(TsOptgroupComponent)
+  public optionGroups!: QueryList<TsOptgroupComponent>;
+
+  /**
+   * Access the overlay pane containing the options
+   */
+  @ViewChild(CdkConnectedOverlay)
+  public overlayDir!: CdkConnectedOverlay;
+
+  /**
+   * Access the panel containing the select options
+   */
+  @ViewChild('panel')
+  public panel!: ElementRef;
+
+
+  /**
+   * GETTERS
+   */
+
+  /**
+   * Whether the select has a value
+   */
+  public get empty(): boolean {
+    return !this.autocompleteFormControl.value.length;
   }
-  public get debounceDelay(): number {
-    return this._debounceDelay;
+
+  /**
+   * Whether the input has focus
+   */
+  public get focused(): boolean {
+    const el = this.inputElement && this.inputElement.nativeElement;
+    return (this.document.activeElement === el) || this.panelOpen;
   }
-  private _debounceDelay = DEFAULT_DEBOUNCE_MS;
 
   /**
-   * A function to output the UI text from the selected item
+   * Determine if the label should float
+   */
+  public get shouldLabelFloat(): boolean {
+    return this.focused || !this.empty;
+  }
+
+
+  /**
+   * INPUTS
+   */
+
+  /**
+   * Define if multiple selections are allowed
+   */
+  @Input()
+  public allowMultiple = false;
+
+  /**
+   * Define if should allow duplicate selections
+   */
+  @Input()
+  public allowDuplicateSelections = false;
+
+  /**
+   * Define if the panel should reopen after a selection is made
    *
-   * When undefined the full selection object will be used as the display value
+   * NOTE: Though it is technically 're-opening', it happens fast enough so that it doesn't appear to close at all.
+   */
+  @Input()
+  public reopenAfterSelection = false;
+
+
+  /**
+   * Define a function to retrieve the UI value for an option
    */
   @Input()
-  public set displayWith(value: TsAutocompleteFormatterFn<OptionType>) {
+  public set chipFormatUIFn(value: TsAutocompleteFormatterFn) {
     if (!value) {
       return;
     }
 
     if (isFunction(value)) {
-      this.uiFormatFn = value;
+      this._chipFormatUIFn = value;
     } else if (isDevMode()) {
-      throw Error(`TsAutocompleteComponent: 'displayWith' must be passed a function.`);
+      throw Error(`TsSelectComponent: 'chipFormatUIFn' must be passed a 'TsAutocompleteFormatFn'.`);
     }
   }
-  public get displayWith(): TsAutocompleteFormatterFn<OptionType> {
-    return this.uiFormatFn;
+  public get chipFormatUIFn(): TsAutocompleteFormatterFn {
+    return this._chipFormatUIFn;
   }
+  private _chipFormatUIFn!: TsAutocompleteFormatterFn;
+
 
   /**
-   * Define a hint for the input
+   * Define a debounce delay for the query stream
    */
   @Input()
-  public hint: string | undefined;
+  public set debounceDelay(value: number) {
+    this._debounceDelay = coerceNumberProperty(value);
```

Comment:
We should set the fallback to the default debounce: `coerceNumberProperty(value, DEBOUNCE_DELAY)`